imagersite/imager_images/views.py

```diff
@@ -1,16 +1,189 @@
+
+
+@login_required
+def edit_album_view(request, album_id):
+    if request.user != Album.objects.filter(pk=album_id).first().owner:
+        raise PermissionDenied
+    album = Album.objects.filter(id=album_id).first()
+    if request.method == 'POST':
+        form = AlbumForm(request.POST, instance=album)
+        if form.is_valid():
+            new_album = form.save(commit=False)
+            new_album.owner = request.user
+            new_album.save()
+            form.save_m2m()
+            return HttpResponseRedirect('/images/library')
+        else:
+            form.fields['photos'].queryset = Photo.objects.filter(owner=request.user)
+            form.fields['cover'].queryset = Photo.objects.filter(owner=request.user)
+            return render(request, 'edit_album.html',
+                          {'form': form.as_p, 'album_id': album_id})
+    else:
+        form = AlbumForm(instance=album)
+        form.fields['photos'].queryset = Photo.objects.filter(owner=request.user)
+        form.fields['cover'].queryset = Photo.objects.filter(owner=request.user)
+        return render(request, 'edit_album.html',
+                      {'form': form.as_p, 'album_id': album_id})
+
+
+@login_required
+def edit_photo_view(request, photo_id):
+    if request.user != Photo.objects.filter(pk=photo_id).first().owner:
+        raise PermissionDenied
+    photo = Photo.objects.filter(id=photo_id).first()
+    if request.method == 'POST':
+        picform1 = PhotoForm(request.POST, request.FILES, instance=photo)
+        picform2 = LocationForm(request.POST)
+        if picform1.is_valid() and picform2.is_valid():
+            new_photo = picform1.save(commit=False)
+            new_photo.owner = request.user
+            new_photo.geom = picform2.cleaned_data['point']
+            new_photo.save()
+            return HttpResponseRedirect('/images/library')
+        else:
+            return render(request, 'edit_photo.html',
+                          {'form1': picform1.as_p,
+                           'form2': picform2,
+                           'album_id': photo_id})
+    else:
+        picform1 = PhotoForm(instance=photo)
+        picform2 = LocationForm()
+        # picform2.fields['point'] = photo.geom
+        return render(request, 'edit_photo.html',
+                      {'form1': picform1.as_p,
+                       'form2': picform2,
+                       'photo_id': photo_id,
+                       'loc': photo.geom})
+
+
+def p_geoview(request, photo_id):
+    """Geo data for photo"""
+    owner = Photo.objects.get(pk=photo_id).owner
+    privacy = Photo.objects.get(pk=photo_id).privacy
+
+    if (request.user != owner) and (privacy == 'Private'):
+        raise PermissionDenied
+    else:
+        photo_lst = Photo.objects.filter(pk=photo_id).all()
+        return HttpResponse(GeoJSONSerializer().serialize(
+            photo_lst, use_natural_keys=True, with_modelname=False,
+            properties=['geom']))
```

Comment:
I'm just noticing this. It doesn't happen when running locally.

On Fri, Aug 7, 2015 at 3:04 PM, Cris Ewing <notifications@github.com> wrote:

> In imagersite/imager_images/views.py
> <https://github.com/tpeek/django-imager/pull/12#discussion_r36565525>:
>
> > +                       'photo_id': photo_id,
> > +                       'loc': photo.geom})
> > +
> > +
> > +def p_geoview(request, photo_id):
> > +    """Geo data for photo"""
> > +    owner = Photo.objects.get(pk=photo_id).owner
> > +    privacy = Photo.objects.get(pk=photo_id).privacy
> > +
> > +    if (request.user != owner) and (privacy == 'Private'):
> > +        raise PermissionDenied
> > +    else:
> > +        photo_lst = Photo.objects.filter(pk=photo_id).all()
> > +        return HttpResponse(GeoJSONSerializer().serialize(
> > +            photo_lst, use_natural_keys=True, with_modelname=False,
> > +            properties=['geom']))
>
> something in here is raising a 500 error when called via ajax from your
> front-end up on AWS. What is it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tpeek/django-imager/pull/12/files#r36565525>.
>